covid19-dashboard/main.py

```diff
@@ -11,181 +11,110 @@
-        'cases': (covid.get_cases_increase_pct_for_given_dates(region=region, most_recent_date=date,
-                                                               time_delta=30)).to_dict(),
-        'deaths': (covid.get_deaths_increase_pct_for_given_dates(region=region, most_recent_date=date,
-                                                                 time_delta=30)).to_dict()
-    }
-
-    print("Getting absoluteCumulative Data")
-    absoluteCumulative = {
-        'cases': covid.get_cumulative_cases_for_given_date(region=region, date=date).to_dict(),
-        'deaths': covid.get_cumulative_deaths_for_given_date(region=region, date=date).to_dict()
-    }
-
-    print("Getting cumulativePerCapita Data")
-    cumulativePerCapita = {
-        'cases': (covid.get_cumulative_cases_per_capita_for_given_date(region=region, date=date)).to_dict(),
-        'deaths': (covid.get_cumulative_deaths_per_capita_for_given_date(region=region, date=date)).to_dict()
-    }
-
-    return {
-        "cumulativeTimeSeries": cumulativeTimeSeries,
-        "daily": daily,
-        "dailyPerCapita": dailyPerCapita,
-        "dailyIncrease": dailyIncrease,
-        "weekly": weekly,
-        "weeklyPerCapita": weeklyPerCapita,
-        "weeklyIncrease": weeklyIncrease,
-        "monthly": monthly,
-        "monthlyPerCapita": monthlyPerCapita,
-        "monthlyIncrease": monthlyIncrease,
-        "absoluteCumulative": absoluteCumulative,
-        "cumulativePerCapita": cumulativePerCapita
-    }
+@cache.cached(timeout=50000)
+@app.route("/get-places")
+def get_places():
+    """Returns a dictionary of dcid->(name, type)
+    where name is the name of the region and type is the type of region."""
+    response: dict = send_request(req_url='https://api.datacommons.org/node/property-values',
+                                  req_json={"dcids": ['NYT_COVID19_County', 'NYT_COVID19_State'],
+                                            "property": "member"},
+                                  post=True,
+                                  compress=False,
+                                  api_key=API_KEY)
+
+    counties: dict = {county['dcid']: (county['name'], 'county') for county in response['NYT_COVID19_County']['out']}
+    states: dict = {state['dcid']: (state['name'], 'state') for state in response['NYT_COVID19_State']['out']}
+    return {**counties, **states}
+
+
+@cache.cached(timeout=50000)
+@app.route("/get-total-cases")
+def get_total_cases():
+    """Returns the total Cumulative Cases of each region per day. day->dcid->cases"""
+    dcids: list = list(get_places().keys())
+    return get_covid_data(dcids, 'NYTCovid19CumulativeCases')
+
+
+@cache.cached(timeout=50000)
+@app.route("/get-total-deaths")
+def get_total_deaths():
+    """Returns the total Cumulative Deaths of each region. day->dcid->deaths"""
+    dcids: list = list(get_places().keys())
+    return get_covid_data(dcids, 'NYTCovid19CumulativeDeaths')
+
+
+@cache.cached(timeout=50000)
```

Comment:
Can make the cache time for get_population and get_places very long, say 3600*24*10, 10 days, since they won't change.

Add make the cache time for get actual stats 3600 like an hour, so will not miss new data update

Reply:
Fixed.